c/src/ml-api-service-offloading.c

```diff
@@ -447,24 +453,44 @@ _mlrs_process_remote_service (nns_edge_data_h data_h, void *user_data)
       g_byte_array_free (array, TRUE);
       break;
     }
-    case ML_REMOTE_SERVICE_TYPE_PIPELINE_RAW:
+    case ML_SERVICE_OFFLOADING_TYPE_PIPELINE_RAW:
       ret = ml_service_pipeline_set (service_key, (gchar *) data);
       if (ML_ERROR_NONE == ret) {
         event_type = ML_SERVICE_EVENT_PIPELINE_REGISTERED;
       }
       break;
+    case ML_SERVICE_OFFLOADING_TYPE_REPLY:
+    {
+      ret = _ml_information_create (&info_h);
+      if (ML_ERROR_NONE != ret) {
+        _ml_error_report_return (ret, "Failed to create information handle. ");
+      }
+      ret = _ml_information_set (info_h, "data", (void *) data, NULL);
+      if (ML_ERROR_NONE != ret) {
+        _ml_error_report ("Failed to set data information.");
+        goto done;
+      }
+      event_type = ML_SERVICE_EVENT_REPLY;
+      break;
+    }
     default:
       _ml_error_report ("Unknown service type or not supported yet. "
           "Service num: %d", service_type);
       break;
   }
 
-  if (remote_s && event_type != ML_SERVICE_EVENT_UNKNOWN) {
-    if (remote_s->event_cb) {
-      remote_s->event_cb (event_type, NULL, remote_s->user_data);
+  if (mls && event_type != ML_SERVICE_EVENT_UNKNOWN) {
+    if (mls->cb_info.cb) {
+      mls->cb_info.cb (event_type, info_h, mls->cb_info.pdata);
     }
   }
 
+done:
+  if (info_h) {
+    ret = ml_information_destroy (info_h);
```

Comment:
unnecessary checking ret code here, call ml_information_destroy (info_h); and return previous error code.